buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -286,11 +286,11 @@ public ByteBuf ensureWritable(int minWritableBytes) {
     final void ensureWritable0(int minWritableBytes) {
         final int writerIndex = writerIndex();
         final int targetCapacity = writerIndex + minWritableBytes;
-        if (targetCapacity <= capacity()) {
+        if (targetCapacity >= 0 && targetCapacity <= capacity()) {
```

Comment:
Since this could be on hot code paths I'm curious whether using `&` instead of `&&` here would make any noticeable difference performance-wise

Reply:
I'm not very familiar with optimisation at this level, but I don't think it'd hurt. We'd be evaluating the second condition in almost every case anyway.